### ISSUE TITLE
Promote, Archive or WIP

### ISSUE THREAD

**Issue by meatballs** [anvilistas/anvil-labs#129]
This project should only include work which are currently investigating. For each of the modules, we need to decide whether it's:

* WIP - it's still being worked on and should stay here
* Ready - to be promoted to Anvil Extras
* Archive - We're finished with it and it should move to the (not yet existing) archive repo.

- [ ] Atomic
- [ ] Kompot
- [ ] Non Blocking
- [ ] Web Worker
- [ ] Zod

**Comment by s-cork** [comment]
my opinions:
- Atomic - still WIP - I'd like to play with the internals, I'm mostly waiting on the progress of a javascritpt framework I follow
- Non-blocking - Ready
- Web-Worker - WIP - too experimental for anvil-extras
- Kompot - used by Web-Worker - I'd say archive otherwise
- Zod - feature complete - Ready 

But I don't really have production apps - so I can't say that they're production ready or not
Or how useful they are to others

**Comment by meatballs** [comment]
Lgtm

**Comment by s-cork** [comment]
any thoughts @hugetim , @jshaffstall ?

**Comment by jshaffstall** [comment]
Non-blocking is definitely something I'd love to use in production.  I haven't played with it as it currently exists, though, so can't say whether it's ready.  But I do have specific situations in my production app where it'd be useful, though not critical.  

Kompot has the the multi-server call wrapping.  I can't say I'd ever use the serialization features of it, but the server call wrapping would be a useful utility.  It allows one to design server functions based on each having a single responsibility, but still execute a bunch in a single server interaction, which I think is very important for writing well designed apps.  

I love the idea of Atomic, so definitely want to see it continue.  It's such a change in structure for apps, though, that I haven't played with it.

**Comment by hugetim** [comment]
I'm a big fan of Kompot, at least in theory. I haven't used it yet but expect that I would use all aspects of it as part of anvil_extras.

**Comment by meatballs** [comment]
So, promote non-blocking and leave atomic, kompot and web worker here as WIP?

**Comment by jshaffstall** [comment]
Sounds good to me.  

Might be worth pulling the server call batching out of Kompot and into a feature of its own.
